--- a/src/Parser.zig
+++ b/src/Parser.zig
@@ -403,6 +403,19 @@ inline fn parseCsi(input: []const u8, text_buf: []u8) Result {
                 .n = sequence.len,
             };
         },
+        'Z' => {
+            return .{
+                .event = .{
+                    .key_press = .{
+                        .codepoint = Key.tab,
+                        .mods = .{
+                            .shift = true,
+                        },
+                    },
+                },
+                .n = sequence.len,
+            };
+        },
         '~' => {
             // Legacy keys
             // CSI number ~
